### PR TITLE
implemented global error handler

### PR DESCRIPTION
**Pull request by paddybasi**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-2567

### Change description ###
Ensured we don't have unwanted assumptions on error objects

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/accounts/index.ts [hunk 1]
```diff
@@ -3,57 +3,59 @@ import * as express from 'express'
...
         }
-    if (responseStatusCode) {
-        res.status(responseStatusCode).send(accounts)
-    } else {
-        res.send(accounts)
+        accounts.push(account.data)
+      })
+    })
+  } catch (error) {
+    const status = valueOrNull(error, 'status')
```
Comment (udaydenduluri33) [inline api/accounts/index.ts:39]: @paddybasi  can you encapsulate this error handling code to a function?

File: api/healthCheck/index.ts [hunk 2]
```diff
@@ -38,41 +39,41 @@ const healthCheckEndpointDictionary = {
...
+    HealthCheckUtil.manageTAndCFeature(isTandCEnabled, healthCheckEndpointDictionary);
+    const Promises = [];
     if (healthCheckEndpointDictionary[path]) {
         healthCheckEndpointDictionary[path].forEach(endpoint => {
             // TODO: Have health config for this.
-            console.log('healthEndpoints')
-            console.log(healthEndpoints()[endpoint])
-            Promises.push(req.http.get(healthEndpoints()[endpoint]))
-        })
+            console.log('healthEndpoints');
```
Comment (udaydenduluri33) [inline api/healthCheck/index.ts:48]: Can you replace with logger?

File: api/accounts/index.ts (hunk 1 above)
Comment (paddybasi) [reply api/accounts/index.ts:39]: This PR is just to protect against a runtime TypeError. We can (and should!) do a further clean up task later, as @Mo-Lala84 mentioned previously. I would rather replace all of these with custom errors that we can throw, then catch in generic error handler

File: api/healthCheck/index.ts (hunk 2 above)
Comment (paddybasi) [reply api/healthCheck/index.ts:48]: again we're not looking to increase functionality etc here, just merely protect against a possible P1 issue

**Review by udaydenduluri33 — APPROVED** [review]
LGTMp

**Review by Mo-Lala84 — APPROVED** [review]
approved

**Pull request closed without merging**